### PR TITLE
Create a credential before adoption and test it after

### PR DESCRIPTION
**Pull request by afaranha**

### CODE REVIEW

File: tests/roles/development_environment/tasks/main.yaml
Comment (xek) [inline tests/roles/development_environment/tasks/main.yaml:59]: This is not being used

File: tests/roles/keystone_adoption/tasks/main.yaml
Comment (xek) [inline tests/roles/keystone_adoption/tasks/main.yaml:109]: You don't need to register the output as a value

File: tests/roles/development_environment/tasks/main.yaml
Comment (xek) [reply tests/roles/development_environment/tasks/main.yaml:59]: Sorry, my Ctrl+F failed me

**Review by xek — APPROVED** [review]
/lgtm

**Comment by jistr** [comment]
LGTM but i'd like to make sure we're ok with this for the RHEV-based job - cc @archana203 @holser

**Review by jistr — APPROVED** [review]
I think it should be ok for the RHEV job too now that we execute the devlopment_environment role, but another pair of eyes would be good :).

**Comment by archana203** [comment]
I would say we should wait for https://github.com/openstack-k8s-operators/data-plane-adoption/pull/545 to be merged first. 
Then the same user_var should be used in this change as well.

File: tests/roles/development_environment/tasks/main.yaml
Comment (archana203) [inline tests/roles/development_environment/tasks/main.yaml:57]: Use alias openstack="ssh -i {{ edpm_privatekey_path }} -o StrictHostKeyChecking=no ${OS_USER}@${OS_CLOUD_IP} OS_CLOUD=${OS_CLOUD_NAME} openstack"

Comment (archana203) [inline tests/roles/development_environment/tasks/main.yaml:57]: Use alias openstack="ssh -i {{ edpm_privatekey_path }} -o StrictHostKeyChecking=no ${OS_USER}@${OS_CLOUD_IP} OS_CLOUD=${OS_CLOUD_NAME} openstack"

File: tests/roles/development_environment/tasks/main.yaml
Comment (archana203) [inline tests/roles/development_environment/tasks/main.yaml:57]: The user name has been changed to SOURCE_OSP_SSH_USER: "{{ source_osp_ssh_user }}"

File: tests/roles/development_environment/tasks/main.yaml
Comment (archana203) [inline tests/roles/development_environment/tasks/main.yaml:61]: Change username here as well ${SOURCE_OSP_SSH_USER}@${OS_CLOUD_IP}

File: tests/roles/development_environment/tasks/main.yaml
Comment (marios) [inline tests/roles/development_environment/tasks/main.yaml:50]: would be nice to call it out in the logs more like 
"create credential for sanity checking its value after adoption"

or similar

**Review by marios — APPROVED** [review]
lgtm
only minor comment on the title so not going to request update for that. 

i think i will merge this it has enough approvals and it seems to work in the jobs here

https://logserver.rdoproject.org/44/544/c00bd562292eb5e429835d99999d563819a1aa0e/github-check/adoption-standalone-to-crc-no-ceph/45e7d90/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/logs/test_minimal_out_2024-08-13T12:15:38EDT.log
```
TASK [keystone_adoption : Verify that pre-adoption credential stills the same] ***
changed: [localhost] => {"changed": true, "cmd": "set -euxo pipefail\n\n\nalias openstack=\"oc exec -t openstackclient -- openstack\"\n${BASH_ALIASES[openstack]} credential show 53daa4e96a234f5a9227d215e7cfab66 -f value -c blob\n", "delta": "0:00:02.854889", "end": "2024-08-13 12:28:02.429436", "failed_when_result": false, "msg": "", "rc": 0, "start": "2024-08-13 12:27:59.574547", "stderr": "+ alias 'openstack=oc exec -t openstackclient -- openstack'\n+ oc exec -t openstackclient -- openstack credential show 53daa4e96a234f5a9227d215e7cfab66 -f value -c blob", "stderr_lines": ["+ alias 'openstack=oc exec -t openstackclient -- openstack'", "+ oc exec -t openstackclient -- openstack credential show 53daa4e96a234f5a9227d215e7cfab66 -f value -c blob"], "stdout": "test", "stdout_lines": ["test"]}
```

**Review by marios — CHANGES_REQUESTED** [review]
please rebase this so we can merge

why are there 7 commits here

**Comment by afaranha** [comment]
Sorry, I tried to squash my commits but it went south, let me fix it

File: tests/roles/development_environment/tasks/main.yaml
Comment (cescgina) [inline tests/roles/development_environment/tasks/main.yaml:53]: do we need this variable? I might be missing something, but this seems to be executing only and `openstack` command

**Comment by cescgina** [comment]
recheck

**Comment by cescgina** [comment]
recheck

**Review by marios — APPROVED** [review]
rebase seems ok now only one commit here

File: tests/roles/development_environment/tasks/main.yaml
Comment (jistr) [reply tests/roles/development_environment/tasks/main.yaml:53]: +1 but looks like the snippets above suffer from the same problem so this can be cleaned up separately i think.

**Pull request merged**
